controllers/storagecluster/storagecluster_controller.go

```diff
@@ -155,6 +155,35 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
+	enqueueFromStorageProfile := handler.EnqueueRequestsFromMapFunc(
+		func(_ context.Context, obj client.Object) []reconcile.Request {
+			spobj, ok := obj.(*ocsv1.StorageProfile)
+			if !ok {
+				spobj.Status.Phase = ocsv1.StorageProfilePhaseFailed
+				return []reconcile.Request{}
+			}
+
+			// Get the StorageCluster object
+			scList := &ocsv1.StorageClusterList{}
+			err := r.Client.List(r.ctx, scList, &client.ListOptions{Namespace: obj.GetNamespace(), Limit: 1})
```

Comment:
@nb-ohad I see this block was changed due to your comments, as there are talks to introduce multiple StorageClasses we are creating a hook for ourselves to fix this at a later point of time, wdyt?

Reply:
@leelavg you mean `StorageCluster`s. We are aware of that and it was decided to postpone taking it into account for this PR, so it's for now out of scope.

Reply:
got it.

Reply:
@leelavg There cannot be two storagecluster CRs in the same namespace and the SP design right now assumes that the SP is related to the SC in the same namespace!

Reply:
Resolving as per @leelavg's comment